src/client/src/App.tsx

```diff
@@ -49,7 +49,9 @@ export const App: React.FC = () => {
             });
         });
     };
-    useEffect(getElementsHook, []);
+    if (process.env.NODE_ENV !== 'test') {
+        useEffect(getElementsHook, [nodeText]);
+    }
```

Comment:
This has the same problem as before. Whenever someone types into the textbox, the app makes a new get request everytime. This seems to have come from some older code and is not present in the main branch. Please fix.

Reply:
By the way, _never_ put useEffect calls into a conditional. While in this case it happens to work, usually it will cause a lot of problems: https://reactjs.org/docs/hooks-rules.html